commit 4f72e09e1a669de12f6f39f5fd2c31eb44b17d76 (refs/changes/29/1429/2)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2017-02-03T15:06:05-08:00 (2 years, 8 months ago)
    
    CNAPI-690 CNAPI needs to be selective with logging request payload
    Reviewed by: Trent Mick <trentm@gmail.com>

diff --git a/config/test.json b/config/test.json
index 2efa467..64b0fec 100644
--- a/config/test.json
+++ b/config/test.json
@@ -34,38 +34,57 @@
     "host": "10.99.99.19"
   },
   "dapi": {
-    "changeDefaults": {
-      "server_spread": "",
-      "filter_headnode": "",
-      "filter_min_resources": "",
-      "filter_large_servers": ""
-    },
-    "allocationDescription": [
-       "pipe", "hard-filter-setup",
-               "hard-filter-running",
-               "hard-filter-invalid-servers",
-               "calculate-ticketed-vms",
-               "calculate-locality",
-               "hard-filter-reserved",
-               "hard-filter-headnode",
-               "hard-filter-vlans",
-               "hard-filter-platform-versions",
-               "hard-filter-traits",
-               "hard-filter-sick-servers",
-               "override-overprovisioning",
-               "calculate-server-unreserved",
-               "hard-filter-overprovision-ratios",
-               "hard-filter-min-ram",
-               "hard-filter-min-cpu",
-               ["or", "hard-filter-reservoir",
-                      "identity"],
-               ["or", "hard-filter-large-servers",
-                      "identity"],
-               "soft-filter-locality-hints",
-               "sort-min-ram",
-               "sort-max-ram",
-               "sort-min-owner",
-               "sort-random",
-               "pick-weighted-random"]
+      "useVmapi": true,
+      "changeDefaults": {
+          "server_spread": "",
+          "filter_headnode": "false",
+          "filter_min_resources": "false",
+          "filter_large_servers": "",
+          "filter_docker_min_platform": "",
+          "filter_vm_limit": "",
+          "disable_override_overprovisioning": "",
+          "overprovision_ratio_cpu": "",
+          "overprovision_ratio_ram": "",
+          "overprovision_ratio_disk": "",
+          "weight_current_platform": "",
+          "weight_next_reboot": "",
+          "weight_num_owner_zones": "",
+          "weight_uniform_random": "",
+          "weight_unreserved_disk": "",
+          "weight_unreserved_ram": ""
+      },
+      "allocationDescription": [
+          "pipe", "hard-filter-setup",
+          "hard-filter-running",
+          "hard-filter-invalid-servers",
+          "hard-filter-volumes-from",
+          "hard-filter-reserved",
+          "hard-filter-vlans",
+          "hard-filter-platform-versions",
+          "hard-filter-traits",
+          "hard-filter-headnode",
+          "override-overprovisioning",
+          "hard-filter-overprovision-ratios",
+          "load-server-vms",
+          "calculate-ticketed-vms",
+          "hard-filter-capness",
+          "hard-filter-vm-count",
+          "hard-filter-sick-servers",
+          "calculate-server-unreserved",
+          "hard-filter-min-ram",
+          "hard-filter-min-cpu",
+          "hard-filter-locality-hints",
+          "hard-filter-owners-servers",
+          ["or", "hard-filter-reservoir",
+              "identity"],
+              ["or", "hard-filter-large-servers",
+                  "identity"],
+                  "soft-filter-locality-hints",
+                  "score-unreserved-ram",
+                  "score-unreserved-disk",
+                  "score-num-owner-zones",
+                  "score-current-platform",
+                  "score-next-reboot",
+                  "score-uniform-random"]
   }
 }
diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index 00dfab3..a4b066e 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -1171,7 +1171,7 @@ VM.dockerBuild = function handlerVmDockerBuild(req, res, next) {
 
 
     req.stash.vm.dockerBuild(
-        { req_id: req.getId(), payload: req.params.payload },
+        { req_id: req.getId(), payload: req.params.payload, log_params: false },
         function (error, result) {
             clearTimeout(timeout);
 
diff --git a/lib/models/server.js b/lib/models/server.js
index b908769..bc4a80c 100644
--- a/lib/models/server.js
+++ b/lib/models/server.js
@@ -2078,7 +2078,10 @@ function (opts) {
     var self = this;
 
     self.log.info('sending task to %s', self.uuid);
-    self.log.info({ params: arguments[0].params }, 'task params');
+
+    if (opts.log_params !== false) {
+        self.log.info({ params: arguments[0].params }, 'task params');
+    }
 
     assert.string(opts.task, 'opts.task');
     assert.object(opts.params, 'opts.params');
diff --git a/lib/models/vm.js b/lib/models/vm.js
index 7c62c10..62cd629 100644
--- a/lib/models/vm.js
+++ b/lib/models/vm.js
@@ -210,7 +210,8 @@ ModelVM.prototype.dockerBuild = function (opts, callback) {
                 callback(error, result);
             },
             req_id: opts.req_id,
-            params: { uuid: self.uuid, payload: opts.payload }
+            params: { uuid: self.uuid, payload: opts.payload },
+            log_params: opts.log_params
         };
 
         servermodel.sendTaskRequest(request);
diff --git a/test/lib/mock.js b/test/lib/mock.js
index 5b68dac..bbbd279 100644
--- a/test/lib/mock.js
+++ b/test/lib/mock.js
@@ -203,6 +203,7 @@ function newApp(callback) {
                 logLevel: 'info',
                 datacenter: config.datacenter_name,
                 cnapi: config.cnapi,
+                dapi: config.dapi,
                 amqp: {
                     host: 'localhost'
                 }
